server/internals/service/service.go

```diff
@@ -227,6 +242,17 @@ func (u *Service) GetClient(id string) (*models.Client, error) {
 	return client, nil
 }
 
+func (u *Service) CheckClient(backendURL string) (*models.Client, error) {
```

Comment:
The interceptor will (likely) map the SP to scheme, subdomain, domain. As such, the public key will also need to be matched to these. Just a heads up.